master/buildbot/reporters/utils.py

```diff
@@ -155,3 +156,11 @@ def getURLForBuild(master, builderid, build_number):
     return prefix + "#builders/%d/builds/%d" % (
         builderid,
         build_number)
+
+
+@renderer
+@defer.inlineCallbacks
+def URLForBuild(props):
+    build = props.getBuild()
+    builder_id = yield build.builder.getBuilderId()
```

Comment:
I suggest to add an accessor in the base class, so any inherited class can easily do ``self.getUrlForBuild()``.